### ISSUE TITLE
[Bug]: ContextualMenu with SearchBox Accessibility Issue aria-required-children

### ISSUE THREAD

**Issue by CDSimpson** [microsoft/fluentui#29127]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 Intel(R) Xeon(R) W-2245 CPU @ 3.90GHz
    Memory: 96.78 GB / 127.59 GB
  Browsers:
    Edge: Spartan (1.1.0), Chromium (1.1.0), ChromiumDev (1.1.0)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

## Actual Behavior
The "ContextualMenu with custom rendered menu list that renders a search box to filter menu items" component is flagged by Accessibility Insights and reports the error seen in the logs.

![Accessibility Insights Failure - Contextual Menu With Search Box](https://github.com/microsoft/fluentui/assets/1693881/2685e5c9-de97-4bba-b72c-00c2be6d0c55)

## Expected Behavior
All permutations of the `<ContextualMenu />` component pass accessibility insights without any reported issues.

## Notes
This seems related to issue #28343, but the difference between that issue and this one is that particular instance has both an `aria-hidden` element (the icon container) _and_ a `role=searchbox` element (the actual searchbox `input` element). The logs call out _both_ of these elements as being problematic.

### Logs

```shell
Title: WCAG 1.3.1: Ensures elements with an ARIA role that require child roles contain them (#id__294-menu)
Tags: Accessibility, WCAG 1.3.1, aria-required-children

Issue: Ensures elements with an ARIA role that require child roles contain them (aria-required-children - https://accessibilityinsights.io/info-examples/web/aria-required-children)

Target application: Fluent UI - Controls - React - ContextualMenu - https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

Element path: #id__294-menu

Snippet: <div id="id__294-menu" class="ms-ContextualMenu-container container-247" tabindex="0" aria-labelledby="id__294" role="menu">

Related paths: 
  .iconContainer-411
  #SearchBox540

How to fix: 
Fix any of the following:
  Element has children which are not allowed: div[aria-hidden], [role=searchbox]
  Element uses aria-busy="true" while showing a loader

Environment: Chrome version 115.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.39.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
Accessibility Insights is correct in that an input or searchbox is technically not an allowed child of a menu. That said, having one as a child doesn't cause significant practical issues for screen reader or keyboard users.

I've added a PR to add warning text to that example on our docs site.
